File: src/lib/api/greetings.ts
import { invoke } from '@tauri-apps/api/tauri';

/**
 * Greet a person by name
 * @param name The name to greet
 * @returns A greeting message from the Rust backend
 */
export async function greet(name: string): Promise<string> {
	return await invoke<string>('greet', { name });
}

File: src-tauri/src/commands/greetings.rs
/// Greets a person by name
/// 
/// # Arguments
/// 
/// * `name` - The name of the person to greet
/// 
/// # Returns
/// 
/// A greeting message as a String
#[tauri::command]
pub fn greet(name: String) -> String {
    format!("Hello, {}! You've been greeted from Rust!", name)
}

